homeassistant/components/cloud/__init__.py

```diff
@@ -134,7 +141,7 @@ def async_active_subscription(hass: HomeAssistant) -> bool:
 @bind_hass
 async def async_create_cloudhook(hass: HomeAssistant, webhook_id: str) -> str:
     """Create a cloudhook."""
-    if not async_is_logged_in(hass):
+    if not async_is_logged_in(hass) or not async_is_connected(hass):
```

Comment:
This is overloading the `CloudNotAvailable` error to mean "not logged in" or "not connected". We should introduce a new error instead.